src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c

```diff
@@ -805,11 +805,14 @@ int xocl_init_mem(struct xocl_drm *drm_p)
 
 	/* Initialize memory stats based on Group topology */
 	err = XOCL_GET_GROUP_TOPOLOGY(drm_p->xdev, group_topo);
-	if (err)
+	if (err) {
+		XOCL_PUT_MEM_TOPOLOGY(drm_p->xdev);
 		return err;
+	}
 
 	if (group_topo == NULL) {
 		err = -ENODEV;
+		XOCL_PUT_MEM_TOPOLOGY(drm_p->xdev);
 		XOCL_PUT_GROUP_TOPOLOGY(drm_p->xdev);
 		return err;
```

Comment:
we are still not releasing the lock. add "goto done"

Reply:
Added